demos/text_detection_demo/cpp_gapi/src/text_recognition.cpp

```diff
@@ -0,0 +1,208 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+#include <fstream>
+
+#include "text_recognition.hpp"
+
+bool fileExists(const std::string& filename) {
+    std::ifstream f(filename.c_str());
+    return f.good();
+}
+bool tryReadVocabFile(const std::string& filename, std::string& alphabet) {
+    if (fileExists(filename)) {
+        alphabet = "";
+        std::ifstream inputFile(filename);
+        if (!inputFile.is_open()) {
+            throw std::runtime_error("Can't open the vocab file: " + filename);
+        }
+        std::string vocabLine;
+        while (std::getline(inputFile, vocabLine)) {
+            if (vocabLine.length() != 1) {
+                throw std::invalid_argument("Lines in the vocab file must contain 1 character");
+            }
+            alphabet += vocabLine;
+        }
+        if (alphabet.empty()) {
+            throw std::logic_error("File is empty: " + filename);
+        }
+        return true;
+    } else {
+        return false;
+    }
```

Comment:
I suggest:
    return true;
}
return false;

Without else we save one line :)

Reply:
OK, nice one :)